app/controllers/admin/configs_controller.rb

```diff
@@ -100,6 +109,17 @@ def bust_relevant_caches
       CacheBuster.bust("/tags/onboarding")
     end
 
+    # Validations
+    def brand_contrast_too_low
+      hex = params[:site_config][:primary_brand_color_hex]
+      hex.present? && WCAGColorContrast.ratio(hex.delete("#"), "ffffff") < 4.5 # a11y
+    end
+
+    def brand_color_not_hex
+      hex = params[:site_config][:primary_brand_color_hex]
+      hex.present? && !hex.match(/\A#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})\z/)
```

Comment:
There's a predicate version of `match` called `match?`.  There are also relevant character classes defined for hex digits:

```
/\h/ - A hexdigit character ([0-9a-fA-F])
/\H/ - A non-hexdigit character ([^0-9a-fA-F])
/[[:xdigit:]]/ - Digit allowed in a hexadecimal number (i.e., 0-9a-fA-F)
```

Using both we'd end up with this
```suggestion
      hex.present? && !hex.match?(/\A#(\h{6}|\h{3})\z/)
```